src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/items/SoulboundRune.java

```diff
@@ -0,0 +1,117 @@
+
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.Sound;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Item;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.Research;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SimpleSlimefunItem;
+import me.mrCookieSlime.Slimefun.Objects.handlers.ItemDropHandler;
+import me.mrCookieSlime.Slimefun.Setup.Messages;
+import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.PlayerProfile;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class SoulboundRune extends SimpleSlimefunItem<ItemDropHandler> {
+    private Research research = null;
+
+    public SoulboundRune(Category category, ItemStack item, String id, RecipeType type, ItemStack[] recipe) {
+        super(category, item, id, type, recipe);
+    }
+
+    @Override
+    public ItemDropHandler getItemHandler() {
+        return (e, p, i) -> {
+            ItemStack item = i.getItemStack();
+            if (SlimefunManager.isItemSimiliar(item, SlimefunItems.RUNE_SOULBOUND, true)) {
+                if (!PlayerProfile.fromUUID(p.getUniqueId()).hasUnlocked(research)) {
+                    Messages.local.sendTranslation(p, "messages.not-researched", true);
+                    return true;
+                }
+
+                Bukkit.getScheduler().scheduleSyncDelayedTask(SlimefunPlugin.instance, () -> {
+                    // Being sure the entity is still valid and not picked up or whatsoever.
+                    if (!i.isValid()) return;
+
+                    Location l = i.getLocation();
+                    Collection<Entity> entites = l.getWorld().getNearbyEntities(l, 1.5, 1.5, 1.5,
+                            entity -> entity.getType() == EntityType.DROPPED_ITEM &&
+                                    !SlimefunManager.isItemSimiliar(((Item) entity).getItemStack(), SlimefunItems.RUNE_SOULBOUND, true)
+                    );
+
+                    ItemStack ench = null;
+                    Item ent = null;
+                    // Collections do not have a #get method so we need to use a for loop.
+                    // We do not use streams for foreach loops as they are more resource consuming.
+                    for (Entity entity: entites) {
+                        ItemStack dropped = ((Item) entity).getItemStack();
+                        if (SlimefunManager.isItemSoulbound(dropped)) return;
+                        ench = ((Item) entity).getItemStack();
+                        ent = (Item) entity;
+                        break;
```

Comment:
Looping and breaking on the first item is really bad-practice.
You should really use .stream().findFirst() for this.

Also: Streams are not resource-intensive, a List completely filled with 1 billion Objects has only proven to be between 1 and 2 milliseconds faster than the Stream.
That's 1-2 divided by 1 billion milliseconds per element and since you are using this to find... 1 element, this argument is just not valid, the difference wouldn't even be measurable.

And especially because you won't be using .stream().forEach() for this but findFirst() instead: It's actually better to use the Stream.

Reply:
Yes but still we are checking if that Item is valid.

Reply:
Read my previous comments...